src/resizing.jl

```diff
@@ -0,0 +1,158 @@
+                            for k = 1:size(A,1)
+                                i_1 = k
+                                out[indx+=1] += half*(@nref $N A i)
+                            end
+                        else
+                            for k = 1:size(A,1)
+                                i_1 = k
+                                tmp = quarter*(@nref $N A i)
+                                out[indx+=1] += tmp
+                                out[indx+strd] = tmp
+                            end
+                        end
+                    end
+                end
+            else
+                threeeighths = convert(eltype(T), 0.375)
+                oneeighth = convert(eltype(T), 0.125)
+                indx = 0
+                if dim == 1
+                    z = convert(T, zero(A[1]))
+                    @inbounds @nloops $N i d->(d==1 ? (1:1) : (1:size(A,d))) d->(j_d = i_d) begin
+                        c = d = z
+                        for k = 1:size(out,1)-1
+                            a = c
+                            b = d
+                            j_1 = 2*k
+                            i_1 = j_1-1
+                            c = convert(T, @nref $N A i)
+                            d = convert(T, @nref $N A j)
+                            out[indx+=1] = oneeighth*(a+d) + threeeighths*(b+c)
+                        end
+                        out[indx+=1] = oneeighth*c+threeeighths*d
+                    end
+                else
+                    fill!(out, zero(T))
+                    strd = stride(out, dim)
+                    stride_1 = 1
+                    @nexprs $N d->(stride_{d+1} = stride_d*size(out,d))
+                    @nexprs $N d->offset_d = 0
+                    peakfirst = true
+                    @inbounds @nloops $N i d->(d==1?(1:1):(1:size(A,d))) d->(if d==dim; peakfirst=isodd(i_d); offset_{d-1} = offset_d+(div(i_d+1,2)-1)*stride_d; else; offset_{d-1} = offset_d+(i_d-1)*stride_d; end) begin
+                        indx = offset_0
+                        if peakfirst
+                            for k = 1:size(A,1)
+                                i_1 = k
+                                tmp = @nref $N A i
+                                out[indx+=1] += threeeighths*tmp
+                                out[indx+strd] += oneeighth*tmp
+                            end
+                        else
+                            for k = 1:size(A,1)
+                                i_1 = k
+                                tmp = @nref $N A i
+                                out[indx+=1] += oneeighth*tmp
+                                out[indx+strd] += threeeighths*tmp
+                            end
+                        end
+                    end
+                end
+            end
+        end
+    end
+end
+
+restrict_size(len::Integer) = isodd(len) ? (len+1)>>1 : (len>>1)+1
+
+## imresize
+function imresize!(resized, original::AbstractMatrix)
```

Comment:
Eventually needs to be made multidimensional. (Again, not your responsibility unless you want to tackle that.) Really we should use Interpolations, which will give us much more flexibility with the interpolation scheme and boundary conditions.